arangod/RocksDBEngine/RocksDBPersistedLog.cpp

```diff
@@ -38,50 +41,43 @@
 using namespace arangodb;
 using namespace arangodb::replication2;
 using namespace arangodb::replication2::replicated_log;
+using namespace arangodb::replication2::replicated_state;
 
-RocksDBPersistedLog::RocksDBPersistedLog(
-    replication2::GlobalLogIdentifier id, uint64_t objectId,
-    std::shared_ptr<RocksDBLogPersistor> persistor)
-    : PersistedLog(std::move(id)),
-      _objectId(objectId),
-      _persistor(std::move(persistor)) {}
-
-auto RocksDBPersistedLog::insert(PersistedLogIterator& iter,
-                                 WriteOptions const& options) -> Result {
-  rocksdb::WriteBatch wb;
-  auto res = prepareWriteBatch(iter, wb);
-  if (!res.ok()) {
-    return res;
+void AsyncLogWriteContext::waitForCompletion() {
+  while (true) {
+    auto c = _pendingAsyncOperations.load(std::memory_order_acquire);
+    if (c == 0) {
+      break;
+    }
+    _pendingAsyncOperations.wait(c, std::memory_order_acquire);
```

Comment:
```suggestion
    _pendingAsyncOperations.wait(c, std::memory_order_relaxed);
```

Reply:
I think you're right, but I don't want to change this without checking back with @maierlars, as he authored this part. However, as he's out for a few days, I'd like to discuss this with him after the merge, in order not to delay it further, if that's fine with you.

Reply:
Ok

Reply:
Ok, but in such case maybe add todo?

Reply:
I've opened a ticket. https://arangodb.atlassian.net/browse/CINFRA-648